kickstarter/android-oss
# MBL-1556: Add network calls to confirm address and requery the PPO card list

# 📲 What

- Requery ppo list when confirm address call is successful
- switch to `schema.graphql` and remove `schema.json`, including the small changes needed with the switch
- Cleanup activity result call back for fix payment and confirm address
- Remove categories from ppo query response, deprecated

# 🤔 Why

ppo! 

# 🛠 How

- Will be using `schema.graph` moving forward
- Confirm address dialog box, tapping confirm sends a mutation request to `CreateOrUpdateBackingAddress`, if that is successful, it then refreshes the ppo list by sending a new query request to `PledgedProjectsOverview`. otherwise we show an error

# 👀 See
Please note -> the "requery" does not dismiss the card because we are using mocked/hardcoded data for example purposes. 

# Success:

https://github.com/user-attachments/assets/ef4356aa-8b93-49c3-a3ee-d31d0c9f6ed9 

# Error:

https://github.com/user-attachments/assets/a1afaa19-c1ba-4efb-8ffc-54164db889a7


# 📋 QA
**Please note -> the requery occurs but does not dismiss the card because we are using mocked/hardcoded data for example and testing purposes.** 

Happy path:
1. Find a `backingID` and `addressID` associated with your account. i used graphiql to find mine
2. In PPOCardFactory.kt, in the `confirmAddressCard()` function, replace the `addressID` and `backingID` values with the ones associate with your account
3. In `GraphqlTransformers`, add a confirm address card to the list, generated by the `PPOCardFactory.confirmAddressCard()` function
4. Run app
5. Navigate to ppo screen. Find confirm address card, tap confirm. Dialog popup should appear
6. In dialog box, tap confirm. Should see loading spinner and success snackbar at the bottom of screen
7. Confirm in logs or app inspection that two calls were made successfully, one to `CreateOrUpdateBackingAddress`, which returns `true`, and one to `PledgedProjectsOverview`

Unhappy path/errors:
1. Make sure the `addressID` and the `backingID` values in the `POCardFactory.confirmAddressCard()` function are both not real ids, can be a random string of characters
2. In `GraphqlTransformers`, add a confirm address card to the list, generated by the `PPOCardFactory.confirmAddressCard()` function
3. Run app
4. Navigate to ppo screen. Find confirm address card, tap confirm. Dialog popup should appear
5. In dialog box, tap confirm. Should see loading spinner and error snackbar at the bottom of screen
6. Confirm in logs or app inspection that only one calls was made to `CreateOrUpdateBackingAddress` with a response of `false`

# Story 📖

[MBL-1556: Add network calls to confirm address and requery the PPO card list](https://kickstarter.atlassian.net/browse/MBL-1556)
